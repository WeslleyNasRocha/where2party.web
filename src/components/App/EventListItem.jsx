import React from 'react';
import { Link } from 'react-router-dom';

const EventListItem = ({ id, description, titulo }) => (
  <div>
    <Link to={`/edit/${id}`}>
      <h3>{titulo}</h3>
    </Link>
    <p>{description}</p>
  </div>
);

export default EventListItem;
