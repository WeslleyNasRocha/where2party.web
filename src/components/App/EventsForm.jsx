import React from 'react';
import { Formik, Field, Form } from 'formik';
import Yup from 'yup';

const EventsForm = props => (
  <Formik
    initialValues={{ email: '', password: '' }}
    onSubmit={({ titulo, description }) => {
      props.submit({ titulo, description });
    }}
    validationSchema={Yup.object().shape({
      titulo: Yup.string()
        .min(5)
        .required(),
      description: Yup.string()
        .min(10)
        .max(250),
    })}
    render={({ touched, errors }) => {
      console.log(errors);
      return (
        <Form>
          <div className="box-body">
            <div className="form-group">
              <label htmlFor="Titulo" className="col-sm-2 control-label">
                Titulo
              </label>

              <div className="col-sm-10">
                <Field
                  type="Text"
                  name="titulo"
                  className="form-control"
                  id="Titulo"
                  placeholder="Titulo"
                />
              </div>
              {touched.titulo &&
                errors.titulo && (
                  <div style={{ marginTop: 10 }} className="alert alert-danger" role="alert">
                    <p>{errors.titulo}</p>
                  </div>
                )}
            </div>
            <div className="form-group">
              <label htmlFor="description" className="col-sm-2 control-label">
                Descrição
              </label>
              <div className="col-sm-10">
                <Field
                  name="description"
                  id="description"
                  type="text"
                  className="form-control"
                  placeholder="Descrição"
                />
              </div>
              {touched.description &&
                errors.description && (
                  <div style={{ marginTop: 10 }} className="alert alert-danger" role="alert">
                    <p>{errors.description}</p>
                  </div>
                )}
            </div>

            <div className="box-footer">
              <button type="submit" className="btn btn-info btn-block">
                save
              </button>
            </div>
          </div>
        </Form>
      );
    }}
  />
);

export default EventsForm;
