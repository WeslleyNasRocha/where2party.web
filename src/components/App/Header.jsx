import React from 'react';
import { NavLink, Link } from 'react-router-dom';

const Header = () => (
  <header>
    <h1>Where 2 Party</h1>
    <nav>
      <div>
        <Link to="/">
          <p>Dashboard</p>
        </Link>
        <Link to="/events">
          <p>Eventos</p>
        </Link>
        <Link to="/addEvent">
          <p>Adicionar Evento</p>
        </Link>
        <Link to="/report">
          <p>Relatorio</p>
        </Link>
      </div>
    </nav>
  </header>
);

export default Header;
