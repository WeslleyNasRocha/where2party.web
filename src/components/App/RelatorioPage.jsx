import React from 'react';

export const RelatorioPage = () => (
  <div>
    <p>RelatorioPage</p>
  </div>
);

export default RelatorioPage;
