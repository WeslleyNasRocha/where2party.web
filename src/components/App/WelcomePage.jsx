import React from 'react';

export const WelcomePage = () => (
  <div>
    <p>Bem vindo ao seu dashboard de eventos</p>
  </div>
);

export default WelcomePage;
