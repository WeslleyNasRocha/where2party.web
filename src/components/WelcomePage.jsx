import React from 'react';

const WelcomePage = () => (
  <div>
    <p>Bem vindo ao meu portfolio</p>
  </div>
);

export default WelcomePage;
