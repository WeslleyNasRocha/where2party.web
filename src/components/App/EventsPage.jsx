import React from 'react';
import { connect } from 'react-redux';

import EventListItem from './EventListItem';

const EventsPage = props => (
  <div>
    {console.log(props.events)}
    {props.events.lenght === 0 ? (
      <p>sem eventos</p>
    ) : (
      props.events.map(event => <EventListItem key={event.id} {...event} />)
    )}
  </div>
);

const mapStateToProps = state => ({
  events: state.events,
});

export default connect(mapStateToProps)(EventsPage);
