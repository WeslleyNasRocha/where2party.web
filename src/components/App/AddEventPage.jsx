import React, { Component } from 'react';
import EventsForm from './EventsForm';
import { connect } from 'react-redux';
import { addEvent } from '../../actions/events';

class AddEventPage extends Component {
  onSubmit = event => {
    this.props.addEvent(event);
    this.props.history.push('/');
  };
  render() {
    return (
      <div className="col-md-6">
        <div className="box box-success">
          <div className="box-header with-border">
            <h2 className="box-title">Add Event Page</h2>
          </div>
          <EventsForm submit={this.onSubmit} />
        </div>
      </div>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  addEvent: event => dispatch(addEvent(event)),
});

export default connect(undefined, mapDispatchToProps)(AddEventPage);
