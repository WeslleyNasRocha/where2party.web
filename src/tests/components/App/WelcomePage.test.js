import React from 'react';
import { shallow } from 'enzyme';

import { WelcomePage } from '../../../components/App/WelcomePage';

test('should render WelcomePage correctly', () => {
  const wrapper = shallow(<WelcomePage />);

  expect(wrapper).toMatchSnapshot();
});
