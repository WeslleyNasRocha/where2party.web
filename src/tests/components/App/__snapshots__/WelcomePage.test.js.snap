// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render WelcomePage correctly 1`] = `
<div>
  <p>
    Bem vindo ao seu dashboard de eventos
  </p>
</div>
`;
